### PR TITLE
docs: rewrite readme

### LINKED ISSUE #8436: docs: rewrite the README
### Please describe the issue

documentation can always be improved

while doing #8251, it would be good to update the README. also some recent VoDa internal feedback

opening this for tracking + any community input

- clear and concise explanation of Ibis
- at least one diagram/picture 
- notes on use cases and who it's for
- example code

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by lostmygithubaccount**
## Description of changes

opening for discussion/feedback. this:

- makes the README cleaner
- quickly gets to bullet points for "Why", then links out to docs for further reading
- quickly gets to code example
- other edits

we should also IMO update the GitHub description to "the portable Python dataframe library" -- the "flexibility of Python with the scale and performance of modern SQL" is in the REAMDE now 

feedback welcome

## Issues closed

closes #8436 

cc: @chiphuyen @zhenzhongxu @ianmcook for feedback (in addition to the Ibis team who will see this PR anyway)

### CODE REVIEW

**Comment by lostmygithubaccount** [comment]
probably worth going here to review: https://github.com/lostmygithubaccount/ibis/tree/readme

File: README.md [hunk 1]
```diff
@@ -1,188 +1,194 @@
...
-- [PostgreSQL](https://ibis-project.org/backends/postgresql/)
-- [SQL Server](https://ibis-project.org/backends/mssql/)
-- [SQLite](https://ibis-project.org/backends/sqlite/)
-- [Snowflake](https://ibis-project.org/backends/snowflake) (experimental)
-- [Trino](https://ibis-project.org/backends/trino/) (experimental)
+- Fast local dataframes with the default DuckDB backend
+- Lazy (by default) or interactive execution
+- [Mix Python dataframe and SQL code](#python--sql-better-together)
+- Use the same dataframe API for [20+ backends](#backends)
+- Iterate locally and deploy remotely by [changing a single line of code](#portability)
```
Comment (chiphuyen) [inline README.md:19]: should we add an example to demonstrate this? e.g. 
1. start with local experimentation
2. change 1 line of code to deploy remotely

File: README.md [hunk 2]
```diff
@@ -1,188 +1,194 @@
...
 
-The list of supported backends is continuously growing. Anyone can get involved
-in adding new ones! Learn more about contributing to ibis in our contributing
-documentation at https://github.com/ibis-project/ibis/blob/main/docs/CONTRIBUTING.md
+See the documentation on ["Why Ibis?"](https://ibis-project.org/why) to learn more.
```
Comment (chiphuyen) [inline README.md:21]: should we add a graph in readme to explain what ibis is conceptually?

**Review by chiphuyen — COMMENTED** [review]
This looks so much better. Thanks @lostmygithubaccount

File: README.md [hunk 3]
```diff
@@ -1,188 +1,194 @@
...
-- [MySQL](https://ibis-project.org/backends/mysql/)
-- [Oracle](https://ibis-project.org/backends/oracle/) (experimental)
-- [Pandas](https://ibis-project.org/backends/pandas/)
-- [Polars](https://ibis-project.org/backends/polars/) (experimental)
-- [PostgreSQL](https://ibis-project.org/backends/postgresql/)
-- [SQL Server](https://ibis-project.org/backends/mssql/)
-- [SQLite](https://ibis-project.org/backends/sqlite/)
-- [Snowflake](https://ibis-project.org/backends/snowflake) (experimental)
-- [Trino](https://ibis-project.org/backends/trino/) (experimental)
+- Fast local dataframes with the default DuckDB backend
```
[the remark this answers is not in the document]
Comment (lostmygithubaccount) [reply README.md:15]: maybe? I'm open to changing it, but I think this serves a few purposes:

- DuckDB hype: DuckDB is awesome, people love it, and Ibis provides a Pythonic dataframe on top by default
- explain the speed -- instead of wondering how Ibis can be so fast (did it implement its own engine?) a user can realize it's from DuckDB
- hint at portability/decoupling from API from engine -- if a user grasps that the Ibis API compiles down to DuckDB SQL, they can hopefully grasp how other backends work

File: README.md [hunk 4]
```diff
@@ -1,188 +1,194 @@
...
-The list of supported backends is continuously growing. Anyone can get involved
-in adding new ones! Learn more about contributing to ibis in our contributing
-documentation at https://github.com/ibis-project/ibis/blob/main/docs/CONTRIBUTING.md
+See the documentation on ["Why Ibis?"](https://ibis-project.org/why) to learn more.
 
-## Installation
+## Getting started
 
-Install Ibis from PyPI with:
+You can `pip install` Ibis with a backend and example data:
```
[the remark this answers is not in the document]
Comment (lostmygithubaccount) [reply README.md:25]: generally no, and generally you wouldn't want to install `examples`. this installs `gcsfs` and `pins`, the latter being a library Ibis uses for examples which are then read from a public GCS bucket

however for the code below -- `t = ibis.examples.penguins.fetch()` -- this is needed. we could throw a CSV/Parquet file somewhere else for this

File: README.md (hunk 1 above)
Comment (lostmygithubaccount) [reply README.md:19]: yes, though this example would necessarily not work without the user having that remote backend setup and getting their connection (credentials and whatnot) right

we could show DuckDB -> MotherDuck as a simple example

we could also just show like DuckDB -> Snowflake/BigQuery and have a warning about "you need to setup your own remote backend/credentials"

File: README.md (hunk 2 above)
Comment (lostmygithubaccount) [reply README.md:21]: yes but I don't have a good one

there is a nice one here: https://voltrondata.com/resources/open-source-standards

we'd need to de-VoDa-fy it. perhaps we can engage the VoDa design team after GTC? graphs/diagrams would also be good in general for the website

I suggest we don't block this update on having a graph though (there isn't one currently)

Comment (cpcloud) [reply README.md:21]: If anything, that Sankey diagram seems like a reasonable start. Agree that this shouldn't block.

File: README.md (hunk 1 above)
Comment (cpcloud) [reply README.md:19]: I don't think a code example of this belongs in the readme. We can simply make the claim and then link to a complete example (like [my bigquery + duckdb arrays blogpost](https://ibis-project.org/posts/backend-agnostic-arrays/)).

File: README.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (cpcloud) [reply README.md:25]: > i'd decouple what is needed for ibis and what's just for demonstration purposes

It's already decoupled, that is the purpose of having a separate `examples` [extra](https://setuptools.pypa.io/en/latest/userguide/dependency_management.html#optional-dependencies).

File: README.md [hunk 5]
```diff
@@ -1,188 +1,194 @@
...
-- [Pandas](https://ibis-project.org/backends/pandas/)
-- [Polars](https://ibis-project.org/backends/polars/) (experimental)
-- [PostgreSQL](https://ibis-project.org/backends/postgresql/)
-- [SQL Server](https://ibis-project.org/backends/mssql/)
-- [SQLite](https://ibis-project.org/backends/sqlite/)
-- [Snowflake](https://ibis-project.org/backends/snowflake) (experimental)
-- [Trino](https://ibis-project.org/backends/trino/) (experimental)
+- Fast local dataframes with the default DuckDB backend
+- Lazy (by default) or interactive execution
+- [Mix Python dataframe and SQL code](#python--sql-better-together)
```
[the remark this answers is not in the document]
Comment (cpcloud) [reply README.md:17]: How about "Compose table operations with inline SQL" or "Mix and match inline SQL code with table operations"

File: README.md [hunk 6]
```diff
@@ -1,188 +1,194 @@
 # Ibis
 
-[![Documentation Status](https://img.shields.io/badge/docs-docs.ibis--project.org-blue.svg)](http://ibis-project.org)
-[![Project Chat](https://img.shields.io/badge/zulip-join_chat-purple.svg?logo=zulip)](https://ibis-project.zulipchat.com)
-[![Anaconda-Server Badge](https://anaconda.org/conda-forge/ibis-framework/badges/version.svg)](https://anaconda.org/conda-forge/ibis-framework)
+[![Documentation status](https://img.shields.io/badge/docs-docs.ibis--project.org-blue.svg)](http://ibis-project.org)
```
Comment (cpcloud) [inline README.md:3]: Any particular reason to change the case? These don't show up anywhere in the code and are really only for documenting what the link is. You could just as well remove the `Documentation Status` text and keep the inner `[]` empty.

File: README.md [hunk 7]
```diff
@@ -1,188 +1,194 @@
...
-- [Oracle](https://ibis-project.org/backends/oracle/) (experimental)
-- [Pandas](https://ibis-project.org/backends/pandas/)
-- [Polars](https://ibis-project.org/backends/polars/) (experimental)
-- [PostgreSQL](https://ibis-project.org/backends/postgresql/)
-- [SQL Server](https://ibis-project.org/backends/mssql/)
-- [SQLite](https://ibis-project.org/backends/sqlite/)
-- [Snowflake](https://ibis-project.org/backends/snowflake) (experimental)
-- [Trino](https://ibis-project.org/backends/trino/) (experimental)
+- Fast local dataframes with the default DuckDB backend
+- Lazy (by default) or interactive execution
```
Comment (cpcloud) [inline README.md:16]: I don't like the "or" here, it reads like two aspects of ibis that are somehow mutually exclusive and also related to each other, neither of which is true.

Laziness is a abstract property of computation, and interactive mode is an Ibis-specific feature that is unrelated to laziness.

File: README.md [hunk 8]
```diff
@@ -1,188 +1,194 @@
...
-Install Ibis from PyPI with:
+You can `pip install` Ibis with a backend and example data:
 
 ```bash
-pip install 'ibis-framework[duckdb]'
+pip install 'ibis-framework[duckdb,examples]'
 ```
 
-Or from conda-forge with:
+> [!TIP]
```
Comment (cpcloud) [inline README.md:31]: Neat, didn't know about this syntax!

File: README.md [hunk 9]
```diff
@@ -1,188 +1,194 @@
...
-- countries
-- gdp
-- independence
+>>> ibis.set_backend("duckdb")
+>>> ibis.set_backend("polars")
+>>> ibis.set_backend("datafusion")
 ```
 
-Choose the `countries` table and preview its first few rows
+More conventionally, you'll create a connection object:
```
Comment (cpcloud) [inline README.md:168]:
```suggestion
Typically, you'll create a connection object:
```

File: README.md [hunk 10]
```diff
@@ -1,188 +1,194 @@
...
 ```
 
-Show the 5 least populous countries in Asia
+And work with tables in that backend:
 
 ```python
+>>> t = con.table("penguins")
+```
+
+Or, for most backends, read from common file formats like CSV or Parquet:
```
Comment (cpcloud) [inline README.md:182]:
```suggestion
You can also read from file formats like CSV and Apache Parquet:
```

**Review by cpcloud — CHANGES_REQUESTED** [review]

File: README.md [hunk 11]
```diff
@@ -1,188 +1,194 @@
...
+- [Apache Druid](https://ibis-project.org/backends/druid/)
+- [Apache Flink](https://ibis-project.org/backends/flink)
+- [Apache Impala](https://ibis-project.org/backends/impala/)
+- [Apache PySpark](https://ibis-project.org/backends/pyspark/)
+- [BigQuery](https://ibis-project.org/backends/bigquery/)
+- [ClickHouse](https://ibis-project.org/backends/clickhouse/)
+- [Dask](https://ibis-project.org/backends/dask/)
+- [DuckDB](https://ibis-project.org/backends/duckdb/)
+- [Exasol](https://ibis-project.org/backends/exasol)
+- [HeavyAI](https://github.com/heavyai/ibis-heavyai)
```
Comment (cpcloud) [inline README.md:147]: Can we remove this? HeavyAI 100% **will not** work with any recent version of Ibis.

File: README.md (hunk 6 above)
Comment (lostmygithubaccount) [reply README.md:3]: no it just bothered me

File: README.md (hunk 8 above)
Comment (lostmygithubaccount) [reply README.md:31]: I learned about it recently!

File: README.md (hunk 11 above)
Comment (lostmygithubaccount) [reply README.md:147]: yep!

File: README.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (lostmygithubaccount) [reply README.md:17]: going with changing "Mix" to "Compose" -- I'm worred "table operations" and "inline" could be confusing to a casual reader

File: README.md (hunk 1 above)
Comment (lostmygithubaccount) [reply README.md:19]: added a link to it below

File: README.md (hunk 2 above)
Comment (lostmygithubaccount) [reply README.md:21]: added Sankey

File: README.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (cpcloud) [reply README.md]: Not sure whether we should make decisions about anything based on "casual" readers, but we can leave it at that.

**Review by cpcloud — APPROVED** [review]

**Comment by lostmygithubaccount** [comment]
merging as an improvement, can make further improvements later
